docs/src/interface-definition.md

```diff
@@ -0,0 +1,167 @@
+# Interface Definition
+
+There are three main targets for calculations `potential_energy`, `forces` and [virial](https://en.wikipedia.org/wiki/Virial_stress). 
+
+Individual calls are implemented by overloading `AtomsCalculators` functions
+
+- `AtomsCalculators.potential_energy` for potential energy calculation
+- `AtomsCalculators.forces` for allocating force calculation and/or...
+- `AtomsCalculators.forces!` for non-allocating force calculation
+- `AtomsCalculators.virial` for [virial](https://en.wikipedia.org/wiki/Virial_stress) calculation
+
+You do not need to implement all of these. To implement force calculation you only need to implement either allocating or non-allocating force call - see below for more details on how to implement force calculation.
+
+Each call has two inputs: `AtomsBase.AbstractSystem` compatible structure and a `calculator` that incudes details of the calculation method. Additionally keywords can be give. These can be ignored, but they need to be present in the function definition.
+
+- First input is `AtomsBase.AbstractSystem` compatible structure
+- Second input is `calculator` structure
+- Method has to accept keyword arguments (they can be ignored)
+- Non-allocating force call `force!` has an AbstractVector as the first input, to which the evaluated force values are stored (look for more details below)
+
+Outputs for the functions need to have following properties
+
+- Energy is a subtype of `Number` that has a unit with dimensions of energy (mass * length^2 / time^2)
+- Force output is a subtype of `AbstractVector` with element type also a subtype of AbstractVector (length 3 in 3D) and unit with dimensions of force (mass * length / time^2). With additional property that it can be reinterpret as a matrix
+- Virial is a square matrix (3x3 in 3D) that has units of energy times length
```

Comment:
are you sure about the unit? If I write down the virial formula it looks more or less like 
```
 sum  rij * fij 
```
If rij are A and fij eV/A then virial should be eV ?

Reply:
otoh, virial / volume = stress which scales like area, which means virial should scale like length * energy as you say. I'm confused right now, can you clarify?

Reply:
Yes you are correct here.

Reply:
which one? I'm actually no longer sure ...

Reply:
Unless you mean that I'm correct about being confused ...

Reply:
I didn't have your second message when I posted my response...

And now I am confused too...

But, this should be the non average version. So, force*distance = energy is the correct one? When we don't divide with volume.

Reply:
yes that's correct. stress = force / area. This works.

Reply:
I was just confusing myself. Please change the unit to energy and then we are ok.